dvc/output/base.py

```diff
@@ -389,24 +404,30 @@ def checkout(
                 )
             return None
 
+        obj = self.get_obj(filter_info=filter_info)
+        if not obj and (filter_info and filter_info != self.path_info):
+            # backward compatibility
```

Comment:
Currently, if you try to checkout a nonexistent directory/path in a cached directory, we'll just silently swallow it :( That nees to be fixed, but I didn't to minimise the func test changes.